src/AggregateFunctions/HelpersMinMaxAny.h

```diff
@@ -87,28 +54,25 @@ static IAggregateFunction * createAggregateFunctionArgMinMax(const String & name
     const DataTypePtr & val_type = argument_types[1];
 
     WhichDataType which(res_type);
-#define DISPATCH(TYPE) \
+#define DISPATCH(TYPE, SINGLEVALUEDATATYPE) \
     if (which.idx == TypeIndex::TYPE) \
-        return createAggregateFunctionArgMinMaxSecond<MinMaxData, SingleValueDataFixed<TYPE>>(res_type, val_type);
-    FOR_NUMERIC_TYPES(DISPATCH)
+        return new AggregateFunctionArgMinMax<AggregateFunctionArgMinMaxData<SINGLEVALUEDATATYPE, MinMaxData<SINGLEVALUEDATATYPE>>>(res_type, val_type); // NOLINT
```

Comment:
This is for the case where the value and result types are the same, right? What code handles other cases like `argMax(Int, String)`?

Reply:
Ah, you are absolutely right; I missed the fact that this was handling different types. I'm going to close this and revisit it to see if something else can be done. Sorry for the inconvenience